### PR TITLE
Translate javascript

### PR DESCRIPTION
**Pull request by kahnvex**
This adds javascript string translation to browserifyi18n.

Fixes https://github.com/yola/browserifyi18n/issues/7

### CODE REVIEW

File: lib/javascript-translator.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+var translateJavascript = function(copy, gettext) {
+  var re = /trans\(\s*(?:"([^"]+)"|\'([^\']+)\')\s*\)\s*/g;
+  var msgid, needle, replacement, match, bookend, translated, phrase;
+
+  translated = copy;
+  match = re.exec(copy);
+  while (match) {
+    needle = match[0];
+    msgid = match[1] || match[2];
+    phrase = gettext[msgid] || msgid
```
Comment (snitch) [inline lib/javascript-translator.js]: Missing semicolon.

File: test/browserify-i18n-js-spec.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+var through2 = require('through2');
+var i18n = require('../index');
+var path = require('path');
+
+
+chai.should();
+process.chdir(__dirname);
+
+describe('browserify i18n javascript translation', function() {
+  var options = {
```
Comment (snitch) [inline test/browserify-i18n-js-spec.js]: Missing "use strict" statement.

File: test/browserify-i18n-js-spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+var chai = require('chai');
+var browserify = require('browserify');
+var through2 = require('through2');
```
Comment (snitch) [inline test/browserify-i18n-js-spec.js]: 'through2' is defined but never used.

File: test/browserify-i18n-js-spec.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+var i18n = require('../index');
+var path = require('path');
```
Comment (snitch) [inline test/browserify-i18n-js-spec.js]: 'path' is defined but never used.

**Comment by euwest** [comment]
what about creating a dictionary of extensions => translators, so when passing a file through, you just use the translator stored at the file's extension on the dictionary?

**Comment by kahnvex** [comment]
> what about creating a dictionary of extensions => translators, so when passing a file through, you just use the translator stored at the file's extension on the dictionary?

Sure. Will do after figuring out why these tests are busted.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by kahnvex** [comment]
:bell:

**Comment by euwest** [comment]
:+1:

**Pull request merged**